keylime/ima/ima.py

```diff
@@ -352,8 +354,8 @@ def _process_measurement_list(
                         agentAttestState.set_ima_dm_state(dm_validator.state_dump())
 
             # Keep old functionality for writing the parsed files with hashes into a file
-            if m2w is not None and (type(entry.mode) in [ast.Ima, ast.ImaNg, ast.ImaSig]):
-                hash_value = codecs.encode(entry.mode.digest.bytes, "hex")
+            if m2w is not None and (isinstance(entry.mode, (ast.Ima, ast.ImaNg, ast.ImaSig))):
+                hash_value = codecs.encode(entry.mode.bytes(), "hex")
```

Comment:
I think I tested originally, but when I refactored the AST code and missed changing this part. So yes this code is currently not covered by any test.

Reply:
Entry class has a `mode` field that has a bytes() method, but I am not sure whether that is what is accessed here because there is also a Digest class ...

Reply:
You are right we want the digest of the file not the content of the IMA entry. So I think `entry.mode.digest.hash` should be the correct field.

Reply:
done